receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -0,0 +1,243 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusremotewritereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusremotewritereceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const (
+	receiverFormat = "protobuf"
+)
+
+// PrometheusRemoteWriteReceiver - remote write
+type PrometheusRemoteWriteReceiver struct {
+	params       component.ReceiverCreateSettings
+	host         component.Host
+	nextConsumer consumer.Metrics
+
+	mu         sync.Mutex
+	startOnce  sync.Once
+	stopOnce   sync.Once
+	shutdownWG sync.WaitGroup
+
+	server  *http.Server
+	config  *Config
+	logger  *zap.Logger
+	obsrecv *obsreport.Receiver
+}
+
+// NewReceiver - remote write
+func NewReceiver(params component.ReceiverCreateSettings, config *Config, consumer consumer.Metrics) (*PrometheusRemoteWriteReceiver, error) {
+	zr := &PrometheusRemoteWriteReceiver{
+		params:       params,
+		nextConsumer: consumer,
+		config:       config,
+		logger:       params.Logger,
+		obsrecv: obsreport.NewReceiver(obsreport.ReceiverSettings{
+			ReceiverID:             config.ID(),
+			ReceiverCreateSettings: params,
+		}),
+	}
+	return zr, nil
+}
+
+// Start - remote write
+func (rec *PrometheusRemoteWriteReceiver) Start(_ context.Context, host component.Host) error {
+	if host == nil {
+		return errors.New("nil host")
+	}
+	rec.mu.Lock()
+	defer rec.mu.Unlock()
+	var err = component.ErrNilNextConsumer
+	rec.startOnce.Do(func() {
+		err = nil
+		rec.host = host
+		rec.server, err = rec.config.HTTPServerSettings.ToServer(host, rec.params.TelemetrySettings, rec)
+		var listener net.Listener
+		listener, err = rec.config.HTTPServerSettings.ToListener()
+		if err != nil {
+			return
+		}
+		rec.shutdownWG.Add(1)
+		go func() {
+			defer rec.shutdownWG.Done()
+			if errHTTP := rec.server.Serve(listener); errHTTP != http.ErrServerClosed {
+				host.ReportFatalError(errHTTP)
+			}
+		}()
+	})
+	return err
+}
+
+func (rec *PrometheusRemoteWriteReceiver) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	req, err := remote.DecodeWriteRequest(r.Body)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	ctx := rec.obsrecv.StartMetricsOp(context.Background())
```

Comment:
This should happen at the top of the method so it captures the decoding time.  Also, use the context from the incoming request:

```suggestion
	ctx := rec.obsrecv.StartMetricsOp(r.Context())
```